homeassistant/components/seventeentrack/sensor.py

```diff
@@ -72,39 +59,66 @@
     }
 )
 
+ISSUE_PLACEHOLDER = {"url": "/config/integrations/dashboard/add?domain=seventeentrack"}
+
 
 async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Configure the platform and add the sensors."""
+    """Initialize 17Track import from config."""
 
-    session = aiohttp_client.async_get_clientsession(hass)
-
-    client = SeventeenTrackClient(session=session)
-
-    try:
-        login_result = await client.profile.login(
-            config[CONF_USERNAME], config[CONF_PASSWORD]
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+    )
+    if (
+        result["type"] == FlowResultType.CREATE_ENTRY
+        or result["reason"] == "already_configured"
+    ):
+        async_create_issue(
+            hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            is_fixable=False,
+            breaks_in_ha_version="2024.10.0",
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "17Track",
+            },
+        )
+    else:
+        async_create_issue(
+            hass,
+            DOMAIN,
+            f"deprecated_yaml_import_issue_${result['reason']}",
```

Comment:
Why is there a dollar sign in the string?

Reply:
removed